nle/env/tasks.py

```diff
@@ -200,27 +181,33 @@ class NetHackGold(NetHackScore):
     """
 
     def __init__(self, *args, **kwargs):
-        options = kwargs.pop(
-            "options",
-            (
-                "windowtype:rl",
-                "color",
-                "showexp",
-                "nobones",
-                "autopickup",
-                "pickup_types:$",
-            ),
-        )
+        options = kwargs.pop("options", None)
+
+        if options is None:
+            # Copy & swap out "pickup_types".
+            options = []
+            for option in nethack.NETHACKOPTIONS:
+                if option.startswith("pickup_types"):
+                    options.append("pickup_types:$")
+                    continue
+                options.append(option)
 
         super().__init__(*args, options=options, **kwargs)
 
-    def _reward_fn(self, last_response, response, end_status):
+    def _reward_fn(self, last_observation, observation, end_status):
         """Difference between previous gold and new gold."""
         del end_status  # Unused
+        if not self.env.in_normal_game():
+            # Before game started or after it ended stats are zero.
+            return 0.0
+
+        old_blstats = last_observation[self._blstats_index]
+        blstats = observation[self._blstats_index]
+
+        old_gold = old_blstats[14]
```

Comment:
I wonder whether we should turn this into methods instead of indexing using a hard-coded number.

Reply:
Yes, except that it's measurably faster to call fewer functions ...

Reply:
How about we give this a name though and define a constant for each of the indices, or would this slow down things too?

Reply:
No that would be fine.

My excuse has been shown as phony. :D

Reply:
(But more long term I think it might be a good idea to move the tasks into C++ once they have stabilized.)